[dartsim] Add new port

traversaro opened the pull request:
**Describe the pull request**

- What does your PR fix? This PR adds a new port for the DART (Dynamic Animation and Robotics Toolkit) library, a articulated body simulated library used in Robotics and Graphics (http://dartsim.github.io/). Fix https://github.com/microsoft/vcpkg/issues/10015 .

- Which triplets are supported/not supported? Have you updated the CI baseline?
  As far as I know all the triplet should be supported. 

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

traversaro committed:
[dartsim] Add new port

traversaro commented:
Apparently this need a bit more work, marking it as WIP.

PhoebeHui commented:
The second patch failed to apply, could you update it?

Checking patch cmake/DARTMacros.cmake...
error: while searching for:
    endif()
  else()
    set(HAVE_${variable} FALSE CACHE BOOL "Check if ${variable} found." FORCE)
    if(ARGV3) # version
      message(WARNING "Looking for ${dependency} - NOT found, to use"
                      " ${component}, please install ${dependency} (>= ${ARGV3})")
    else()
      message(WARNING "Looking for ${dependency} - NOT found, to use"
                      " ${component}, please install ${dependency}")
    endif()
    return()
  endif()

error: patch failed: cmake/DARTMacros.cmake:135
error: cmake/DARTMacros.cmake: patch does not apply

traversaro commented:
Yes, I already fixed it locally, I need to push it upstream, but the port is still blocked by https://github.com/microsoft/vcpkg/pull/12795 .

traversaro commented:
> Yes, I already fixed it locally, I need to push it upstream, but the port is still blocked by #12795 .


https://github.com/microsoft/vcpkg/pull/12795 has been merged, I just rebased the PR on the top of the latest master.

traversaro committed:
Update vcpkg.json

traversaro committed:
Update vcpkg.json

traversaro commented:
The PR is now failing due to the error: 
~~~
FAILED: dart/utils/CMakeFiles/dart-utils.dir/XmlHelpers.cpp.o 
/usr/bin/c++  -DBOOST_TEST_DYN_LINK -I/mnt/vcpkg-ci/buildtrees/dartsim/src/v6.9.4-8a922c6dbc.clean -I. -isystem /mnt/vcpkg-ci/installed/x64-linux/share/eigen3/../../include/eigen3 -isystem /mnt/vcpkg-ci/installed/x64-linux/include -fPIC -Wall -Wextra -fPIC -g -fno-omit-frame-pointer -fno-inline-functions -fno-inline-functions-called-once -fno-optimize-sibling-calls -MD -MT dart/utils/CMakeFiles/dart-utils.dir/XmlHelpers.cpp.o -MF dart/utils/CMakeFiles/dart-utils.dir/XmlHelpers.cpp.o.d -o dart/utils/CMakeFiles/dart-utils.dir/XmlHelpers.cpp.o -c /mnt/vcpkg-ci/buildtrees/dartsim/src/v6.9.4-8a922c6dbc.clean/dart/utils/XmlHelpers.cpp
In file included from /mnt/vcpkg-ci/installed/x64-linux/include/boost/lexical_cast/detail/converter_lexical_streams.hpp:63:0,
                 from /mnt/vcpkg-ci/installed/x64-linux/include/boost/lexical_cast/detail/converter_lexical.hpp:54,
                 from /mnt/vcpkg-ci/installed/x64-linux/include/boost/lexical_cast/try_lexical_convert.hpp:44,
                 from /mnt/vcpkg-ci/installed/x64-linux/include/boost/lexical_cast.hpp:32,
                 from /mnt/vcpkg-ci/buildtrees/dartsim/src/v6.9.4-8a922c6dbc.clean/dart/utils/XmlHelpers.cpp:38:
/mnt/vcpkg-ci/installed/x64-linux/include/boost/lexical_cast/detail/inf_nan.hpp:34:10: fatal error: boost/math/special_functions/sign.hpp: No such file or directory
 #include <boost/math/special_functions/sign.hpp>
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
~~~

That is probably due to https://github.com/microsoft/vcpkg/issues/8753 . However, an obvious workaround is to add an explicit dependency on `boost-math`.

traversaro committed:
Update vcpkg.json

traversaro commented:
The CI tests are now running correctly.

PhoebeHui reviewed (approved):
The changes looks good to me, thanks for your contribution!

ras0219-msft commented:
Thanks!

jslee02 commented on ports/dartsim/vcpkg.json:
```diff
@@ -0,0 +1,25 @@
+{
+  "name": "dartsim",
+  "version-string": "9.4.0",
```
Shouldn't this version `6.9.4` to match the one in `portfile.cmake`?

traversaro replied:
Good catch @jslee02, I am not sure how the 9.4.0 ended there, probably when I migrated from the `CONTROL` file to the new `vcpkg.json` I copied an existing file and I forgot to update the version. A fix for this is in https://github.com/microsoft/vcpkg/pull/13550 , let's hope the jump backword in version is not creating problems.

jslee02 replied:
Thanks for the quick patch!
